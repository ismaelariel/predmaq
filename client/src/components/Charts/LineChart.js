import React from "react";

import {Chart as ChartJS} from "chart.js/auto";
import {ArcElement, Tooltip, Legend} from "chart.js";
import {Line} from "react-chartjs-2";

ChartJS.register(ArcElement, Tooltip, Legend);

const LineChart = ({machine, action}) => {
    const data = {
        labels: machine.map((data) => data.ProductID),
        datasets: [
            {
                label: action,
                data: machine.map((data) => action === "Desgaste" ? data.ToolWear : data.RotationalSpeed)
            }
        ]
    };
    
    return(
        <Line
            data={data}
            width={100}
            height={100}
            options={{ maintainAspectRatio: false }}
        />
    );
};

export default LineChart;