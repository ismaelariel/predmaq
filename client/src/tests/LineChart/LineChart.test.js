import React from "react";
import renderer from "react-test-renderer";
import machine from "../../utils/machine.json";

import LineChart from "../../components/LineChart/LineChart";

jest.mock("react-chartjs-2");

it("render linechart view", () => {
    const component = renderer.create(
        <LineChart
            predict={machine.data}
            nameState="Desgaste"
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
