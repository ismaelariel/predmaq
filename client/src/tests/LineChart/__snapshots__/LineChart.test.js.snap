// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render linechart view 1`] = `null`;
